src/operator/contrib/ngraph.cc

```diff
@@ -180,6 +184,10 @@ bool NgraphSubgraphInferShape(const nnvm::NodeAttrs &attrs,
 bool NgraphSubgraphInferType(const nnvm::NodeAttrs &attrs,
                              std::vector<int> *iattr, std::vector<int> *oattr) {
   auto graph = get_ngraph(attrs);
+  CHECK(iattr != nullptr);
+  CHECK(oattr != nullptr);
+  CHECK(iattr->size() == graph->inputs_.size());
+  CHECK(oattr->size() == graph->get_results().size());
```

Comment:
Would it make sense to move the checks regarding `oattr` further down, closer to where `oattr` is actually used?  I think that would better adhere to the coding style you've used elsewhere for where the checks are added.